nncf/torch/compression_method_api.py

```diff
@@ -97,50 +97,26 @@ def distributed(self):
         should be made inside this function.
         """
 
-    def load_state(self, state: Dict[str, object]) -> None:
-        """
-        Loads the compression controller state.
-
-        :param state: Output of `get_state()` method.
-        """
-        self._check_loaded_compression_stage(state)
-        if self.scheduler is not None:
-            self.scheduler.load_state(state['scheduler'])
-
-    def _check_loaded_compression_stage(self, state: Dict[str, object]) -> None:
-        if 'compression_level' in state:
-            if 'compression_stage' not in state:
-                compression_level = state['compression_level']
-                state['compression_stage'] = CompressionLevel.map_legacy_level_to_stage()[compression_level]
-            else:
-                nncf_logger.warning('Both CompressionStage and (legacy) CompressionLevel attributes '
-                                    'are specified in the checkpoint. Proceeding with the value stored '
-                                    'in CompressionStage')
-        if 'compression_stage' in state and self.compression_stage() != state['compression_stage']:
-            nncf_logger.warning('Current CompressionStage ({}) of the compression controller does '
-                                'not correspond to the value found in '
-                                'the checkpoint ({})'.format(self.compression_stage(),
-                                                             state['compression_stage']))
-
-    def get_state(self) -> Dict[str, object]:
+    def get_compression_state(self) -> Dict:
```

Comment:
We have the following declaration
```python
@abstractmethod
def get_compression_state(self) -> CompressionState:
```
from the `nncf.api.compression.py` module. But here we write `Dict` instead of `CompressionState`. Is it expected behavior?

Reply:
In PyTorch user don't need `CompressionState` object, instead it's more straightforward to get Dict and save it:
```python
checkpoint = {
  COMPRESSION_STATE_ATTR: compression_ctrl.get_compression_state(),
}
torch.save(checkpoint)
...
torch.load(checkpoint)
compression_state = checkpoint[COMPRESSION_STATE_ATTR]
create_compressed_model(model, config, compression_state=compression_state)
```
If we are 100% aligned with TF the code will be the following:

```python
checkpoint = {
  COMPRESSION_STATE_ATTR: compression_ctrl.get_compression_state().get_state(),
}
torch.save(checkpoint)
...
torch.load(checkpoint)
compression_state = checkpoint[COMPRESSION_STATE_ATTR]
state = CompressionState()
state.load(compression_state)
create_compressed_model(model, config, compression_state=state)
```

I think we could return different things in PT and TF for the simple user code.
What do you think @vshampor @alexsu52 @AlexKoff88 ?

Reply:
@ljaljushkin  Ok, I understand your concern.

It looks very strange. We introduce a common NNCF API and all framework-specific implementations must follow this API. So, all implementations of the `get_compression_state()` method should return derivatives from the `CompressionState` class. If the PT backend wants to use this state as a dictionary, I suggest defining the `PTCompressionState` class and inheriting it from `dict` and `CompressionState`. Otherwise, I don't see the value of a common NNCF API.

Reply:
Fixed. Now all implementations return `CompressionState`.
Additionally, `PTCompressionController` has a method `get_compression_state_dict` which return Dict.